tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2129,3 +2132,126 @@ def test_upgrade_and_enable_basic_auth(self, base_release: tuple[int, int],
         unique_versions = wait_for_num_versions(self.redpanda, 1)
         assert old_version_str not in unique_versions
         self.check_usage()
+
+
+class VerifiablePandaproxyProducer(threading.Thread):
```

Comment:
This looks great!  Probably a future enhancement to the producer/consumer would be to allow for different data formats to be produced/consumed (e.g. binary and, when PP supports serializing/deserializing schemas, the schema registry types).

Reply:
I'm removing this commit from this PR and moving the NetworkDelay test to #9921

Reply:
I think it should do a bit more strict model-based testing. 

The idea would be to produce to Redpanda and an in-memory data-structure (much the same as now). But then consumers should consume from both Redpanda and the data structure. Consumers can run for a few seconds more than the produce.

The conceptual model is that partitions are queue, and each item in the queue should be consumed at least once (maybe exactly once is a bit too strict).

We can then do checks such as:
* Are consumers consuming in the same order as the produce (per partition).
* Are consumers consuming at least once?
* Has everything been consumed?